source/extensions/resource_monitors/injected_resource/injected_resource_monitor.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "envoy/config/resource_monitor/injected_resource/v2alpha/injected_resource.pb.validate.h"
+#include "envoy/filesystem/filesystem.h"
+#include "envoy/server/resource_monitor.h"
+#include "envoy/server/resource_monitor_config.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ResourceMonitors {
+namespace InjectedResourceMonitor {
+
+class InjectedResourceMonitor : public Server::ResourceMonitor {
```

Comment:
can you add a class description? It would  help a noob find his/her way around the code. In particular I don't know whether this is a ResourceMonitor that's being injected, or an InjectedResource that's being monitored :)

Context on how & when this injection happens would be great.